src/components/teasers/teaser.tsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import styled, { css } from 'styled-components';
+import {
+  TeaserTitle,
+  TeaserTitleLarge,
+  Text,
+  Timestamp,
+  Topline,
+} from '../typography/typography';
+import { Arrow } from '../icons/arrow';
+import { theme } from '../layout/theme';
+import { GatsbyImage, IGatsbyImageData } from 'gatsby-plugin-image';
+import { Link } from '../links/links';
+import { up } from '../breakpoint/breakpoint';
+
+const TeaserContainer = styled.div`
+  cursor: pointer;
+  overflow: hidden;
+
+  .teaser-img {
+    transition: transform 0.2s;
+  }
+
+  &:hover {
+    color: ${theme.palette.text.topline};
+    p {
+      color: ${theme.palette.text.topline};
+    }
+
+    .teaser-img {
+      transform: scale(1.05);
+    }
+  }
+`;
+
+const ToplineContainer = styled.div`
+  display: flex;
+  margin-top: 16px;
+  width: 100%;
+  justify-content: space-between;
+`;
+
+const StyledTeaserTitleLarge = styled(TeaserTitleLarge)<{
+  hideOverflow?: boolean;
+}>`
+  ${(props) =>
+    props.hideOverflow &&
+    css`
+      overflow: hidden;
+      white-space: nowrap;
+      text-overflow: ellipsis;
+    `}
+`;
+const StyledTeaserTitle = styled(TeaserTitle)<{ hideOverflow?: boolean }>`
+  ${(props) =>
+    props.hideOverflow &&
+    css`
+      overflow: hidden;
+      white-space: nowrap;
+      text-overflow: ellipsis;
+    `}
+`;
+
+const TeaserText = styled(Text)<{ hasImage: boolean; hideOverflow?: boolean }>`
+  margin-bottom: 22px;
+  margin-top: ${(props) => (props.hasImage ? '8px' : '16px')};
+
+  ${(props) =>
+    props.hideOverflow &&
+    css`
+      overflow: hidden;
+      text-overflow: ellipsis;
+      display: -webkit-box;
+      -webkit-line-clamp: 3;
+      -webkit-box-orient: vertical;
+
+      ${up('md')} {
+        -webkit-line-clamp: 2;
+      }
+    `}
+`;
+
+const ImageContainer = styled.div`
+  overflow: hidden;
+`;
+
+const TeaserImage = styled(GatsbyImage)`
+  margin: 0;
+  width: 100%;
+`;
+
+const Illustration = styled.div`
+  margin-bottom: 24px;
+`;
+
+interface TeaserProps {
+  title: string;
+  topline?: string;
+  timestamp?: string;
+  hideOverflow?: boolean;
+  icon?: JSX.Element;
+  image?: IGatsbyImageData;
+  linkTo: string;
+}
+
+export const Teaser: React.FC<TeaserProps> = ({
+  topline,
+  title,
+  timestamp,
+  icon,
+  image,
+  linkTo,
+  hideOverflow,
+  children,
+}) => {
+  const largeTitle = !image && !icon;
+  return (
+    <TeaserContainer>
+      <Link to={linkTo}>
+        <Illustration>{icon}</Illustration>
+        {image && (
+          <ImageContainer>
+            <TeaserImage className="teaser-img" alt={''} image={image} />
+          </ImageContainer>
```

Comment:
I think you can never combine icon and image. The easiest way to prevent this is to offer a single prop where you can pass in either an icon or an image? I already commented on the image prop being not ideal (it's a gatsby specific data structure, while it could be a whole image component).

So one prop named `cover` or such could bring in the illustration or a gatsby image for example. Those are brief ideas while writing the PR and I'm happy to discuss this. Don't take my opinion as the final truth to implement 🤓 Thank you!

Reply:
With only one property the component would definitely look cleaner 👍
However, some spacing changes depending on whether an icon or an image is used. Is there a best practice to find out in the component which type it is?

Reply:
I can very well imagine that this is a mistake in the designs. I don't see a reason to have different spacings depending on whether the media element is an illustration or an image 🤷 
Maybe you can ask Eric to double check and confirm the spacings?

Reply:
Full agree, this relates to my other comment about the margin I think?